src/Microsoft.TestPlatform.CrossPlatEngine/TestSession/TestSessionPool.cs

```diff
@@ -147,7 +149,13 @@ public virtual bool KillSession(TestSessionInfo testSessionInfo, IRequestData re
         try
         {
             // Deque an actual proxy to do work.
-            return sessionManager.DequeueProxy(source, runSettings);
+            var proxy = sessionManager.DequeueProxy(source, runSettings);
+
+            // Make sure we use the per-request request data instead of the request data used when
+            // creating the test session.
```

Comment:
Please make this sure why need to make sure we use them here.

Reply:
Done, thanks !